### PR TITLE
Add optional warmup step for training.

### LINKED ISSUE #300: Warmup step
Add a `warmup_duration` and `warmup_lr` options to `learn`. `warmup_duration` specifies how many positions (batches?) are performed with `warmup_lr` as the learning rate instead of `lr`.

### PR DESCRIPTION
**Pull request by Sopel97**
Specified with `warmup_epochs`, uses `warmup_lr`.
The purpose is to put the net into a somewhat stable state so that the gradients are not as high during the early stages of the training and don't "accidentally" break the net.

Default values preserve the old behaviour.

Sample output:
```
INFO (initialize_training): Performing random net initialization.
Finished initialization.
info string NNUE evaluation using  enabled
INFO (sfen_reader): Opened file for reading: data/noob_master_d9_1B_0.binpack
INFO (sfen_reader): Opened file for reading: data/noob_master_d9_1B_0.binpack
Doing 10 warmup epochs.
.Finished 1 out of 10 warmup epochs.
.Finished 2 out of 10 warmup epochs.
.Finished 3 out of 10 warmup epochs.
.Finished 4 out of 10 warmup epochs.
.Finished 5 out of 10 warmup epochs.
.Finished 6 out of 10 warmup epochs.
.Finished 7 out of 10 warmup epochs.
.Finished 8 out of 10 warmup epochs.
.Finished 9 out of 10 warmup epochs.
.Finished 10 out of 10 warmup epochs.

PROGRESS (calc_loss): Thu Mar 25 14:42:36 2021, 1000000 sfens, 6266 sfens/second
, epoch 0
  - learning rate = 1
  - startpos eval = 10
  - val_loss       = 0.169671
  - norm = 24825
  - move accuracy = 1.3%
INFO (learn): initial loss = 0.169671
```

Closes #300

### CODE REVIEW

**Comment by nodchip** [comment]
I understand that the first warmup_epochs epochs are trained with learning rate = warmup_lr.  Thank you.
